common/test/acceptance/pages/studio/settings_advanced.py

```diff
@@ -51,12 +51,22 @@ def deprecated_settings_button_text(self):
         """
         return self.q(css=DEPRECATED_SETTINGS_BUTTON_SELECTOR).text[0]
 
-    def check_deprecated_settings_visibility(self):
+    def is_deprecated_setting_visible(self):
         """
             Returns true if deprecated settings are visible
         """
         return self.q(css=DEPRECATED_SETTINGS_SELECTOR).visible
 
+    def toggle_deprecated_settings(self):
+        """
+        Show deprecated Settings
+        """
+        self.q(css=DEPRECATED_SETTINGS_BUTTON_SELECTOR).click()
+        if self.deprecated_settings_button_text == 'Show Deprecated Settings':
```

Comment:
Getting the button text right after clicking it seems prone to race conditions; the button text might not always have updated by this point.  How about saving the button text to a local variable before clicking it, and using that in the`if` condition here?